cmd/fs-v1.go

```diff
@@ -533,22 +513,7 @@ func (fs *FSObjects) GetObjectNInfo(ctx context.Context, bucket, object string,
 		return nil, toObjectErr(err, bucket, object)
 	}
 
-	// Take a rwPool lock for NFS gateway type deployment
-	rwPoolUnlocker := func() {}
-	if bucket != minioMetaBucket {
-		fsMetaPath := pathJoin(fs.fsPath, minioMetaBucket, bucketMetaPrefix, bucket, object, fs.metaJSONFile)
-		_, err = fs.rwPool.Open(fsMetaPath)
```

Comment:
Is this part no longer needed? Wouldn't it cause some breakage in the some setups?

Reply:
> Is this part no longer needed? Wouldn't it cause some breakage in the some setups?

This cause a complex issue for CopyObjectPart, Harsha & I agreed to forget this for the moment, meaning that we are not going to support full consistency in case of shared FS.

Reply:
> This cause a complex issue for CopyObjectPart, Harsha & I agreed to forget this for the moment, meaning that we are not going to support full consistency in case of shared FS.

Basically the problem is that FS locking is based in FD descriptor, which means we need to spread FD number of fs.json hold by GetObjectNInfo to CopyObject() so this latter won't create a new FS lock and just uses the same FD, which is also not so obvious

Reply:
Wait I might have forgotten why it's removed ?

Reply:
> Wait I might have forgotten why it's removed ?

Basically, because we need to spread the FD descriptor of the locked fs.json hold in GetObjectNInfo to other functions like CopyObject() so they ca use the FD to update fs.json, as they cannot call os.Open(fs.json) again because the file is locked

Reply:
Ah yes but this won't make it less consistent. It will make it more stricter.

Reply:
> Ah yes but this won't make it less consistent. It will make it more stricter.

yes right, but currently it causes a problem for CopyObject(), so I thought we can revert this (this is useful only for shared FS) and we do it in another PR

Reply:
One thing to remember here is that we cannot remove consistency guarantees for shared fs mode, I haven't looked at the code or this PR correctly yet. But we cannot undermine existing functionality to be fixed later. Please confirm.

Reply:
The consistency that I am talking about is not the general consistency that we already have, but I am talking about the even stronger consistency that we are trying to implement these days (meaning avoiding returning an error in case of highly concurrent read/write of the same object)